src/ar100/gpio.h

```diff
@@ -0,0 +1,60 @@
+#ifndef GPIO_H
+#define GPIO_H
+
+#include <stdint.h>
+#define R_PIO_BASE  0x01F02C00
+#define PIO_BASE    0x01C20800
+
+enum pin_func {
+  PIO_INPUT,
+  PIO_OUTPUT,
+  PIO_ALT1,
+  PIO_ALT2,
+  PIO_ALT3,
+  PIO_ALT4,
+  PIO_ALT5,
+  PIO_DISABLE
+};
+
+struct gpio_mux {
+  uint32_t pin;
+  uint8_t bank;
+  uint32_t reg;
+};
```

Comment:
It would be preferable to not define ar100 specific definitions in gpio.h.  Instead it's typical to create an "internal.h" header for that purpose.  The reason this is preferable is it can help avoid possible future naming conflicts with the generic code.